commons/ava_testsuite/rpc_manager.go

```diff
@@ -0,0 +1,147 @@
+package ava_testsuite
+
+import (
+	"github.com/kurtosis-tech/ava-e2e-tests/commons/ava_default_testnet"
+	"github.com/kurtosis-tech/ava-e2e-tests/gecko_client"
+	"github.com/palantir/stacktrace"
+	"github.com/sirupsen/logrus"
+	"time"
+)
+
+const (
+	GENESIS_USERNAME = "genesis"
+	GENESIS_PASSWORD = "genesis34!23"
+	ACCEPTED_STATUS = "Accepted"
+)
+
+type RpcManager struct {
+	client *gecko_client.GeckoClient
+	testNet *ava_default_testnet.TestNet
+	rpcUser *RpcUser
+}
+
+func NewRpcManager(
+		client *gecko_client.GeckoClient,
+		testNet *ava_default_testnet.TestNet,
+		username string,
+		password string) *RpcManager {
+	return &RpcManager{
+		client: client,
+		testNet: testNet,
+		rpcUser: NewRpcUser(username, password),
+	}
+}
+
+type RpcUser struct {
+	username string
+	password string
+	payerNonce int
+}
+
+func NewRpcUser(username string, password string) *RpcUser {
+	return &RpcUser{username: username, password: password, payerNonce: 0}
+}
+
+func (rpcUser RpcUser) incrementNonce() int {
+	rpcUser.payerNonce++
+	return rpcUser.payerNonce
+}
+
+/*
+	Creates a new account on the XChain under the username and password.
+	Transfers funds from the genesis account to the new XChain account using the Genesis private key.
+	Returns the new, funded XChain account address.
+ */
+func (rpcManager RpcManager) CreateAndSeedXChainAccountFromGenesis(
+	amount int) (string, error) {
+	time.Sleep(time.Second * 30)
+	client := rpcManager.client
+	username := rpcManager.rpcUser.username
+	password := rpcManager.rpcUser.password
+	_, err := client.KeystoreApi().CreateUser(username, password)
+	if err != nil {
+		stacktrace.Propagate(err, "Could not create user.")
+	}
+	_, err = client.KeystoreApi().CreateUser(GENESIS_USERNAME, GENESIS_PASSWORD)
+	if err != nil {
+		stacktrace.Propagate(err, "Could not create genesis user.")
+	}
+	nodeId, err := client.AdminApi().GetNodeId()
+	if err != nil {
+		return "", stacktrace.Propagate(err, "Could not get node id")
+	}
+	genesisAccountAddress, err := client.XChainApi().ImportKey(
```

Comment:
Bookmark to chat this over: would love to understand how the accounts & keys work; don't feel like I have a strong enough knowledge right now to comment